Admit optional CLI arg specifying config file path

patrickmckenna opened the pull request:
@lra this PR builds on https://github.com/lra/mackup/pull/1143, which seems to have been abandoned? In particular, it:
- Adds validation logic for the user-supplied path, per the request in https://github.com/lra/mackup/pull/1143#discussion_r176891608. (The goal was a more robust check, that the file actually exists, instead of just simple type checking.)
- Automatically handles paths given relative to `$HOME` or the current working directory, along with absolute paths.
- Adds tests of the new behavior, and confirming that the existing default behavior is preserved.

/cc @guilhem

patrickmckenna commented:
@lra friendly bump 😄

lra commented:
Yep, it's a big changes, and review take almost as much time as writing them, but it's in the pipe unless it's closed ;)

patrickmckenna commented:
@lra I know this it's been awhile, but I'm setting up a new machine and running into this issue again 😀It seems to be on others' minds, too, e.g. https://github.com/lra/mackup/issues/1559.

I'm happy to clean up the conflicts and make any other suggested changes, if you think you'll have time to review this? Given the state of the world right now though, I'd certainly understand if you can't get to this for awhile longer.

patrickmckenna commented:
 @tillhainbach yeah, tilde expansion seems like a good idea. I've added that, and fixed the merge conflicts.

While making those changes, Python 2 issues kept popping up. That got me wondering: what's the thinking around whether to continue maintaining backwards compatibility with an officially deprecated version of the language? (I don't mean to get this PR off course--the issue merits a separate thread--but was wondering if there was a quick answer @lra.)

tillhainbach commented:
I think the reasoning behind python2 was that this is the version shipped with Mac OS X/macOS

But yeah, could be worth thinking about a fork (mackup3 ?) which drops python2 support (but that's something to discuss elsewhere I guess)
